packages/block-editor/src/components/block-toolbar/use-has-block-toolbar.js

```diff
@@ -28,24 +27,30 @@ export function useHasBlockToolbar() {
 			// for the toolbar, so we use getBlockSelectionStart
 			// instead of getSelectedBlockClientIds
 			const selectedBlockClientId = getBlockSelectionStart();
-			const isDefaultEditingMode =
-				getBlockEditingMode( selectedBlockClientId ) === 'default';
+
 			const blockType =
 				selectedBlockClientId &&
 				getBlockType( getBlockName( selectedBlockClientId ) );
-			const isToolbarEnabled =
-				blockType &&
-				hasBlockSupport( blockType, '__experimentalToolbar', true );
 
-			if (
-				! isToolbarEnabled ||
-				( ! isDefaultEditingMode && ! hasAnyBlockControls )
-			) {
-				return false;
-			}
-
-			return true;
+			return {
+				isToolbarEnabled:
+					blockType &&
+					hasBlockSupport( blockType, '__experimentalToolbar', true ),
+				isDefaultEditingMode:
+					getBlockEditingMode( selectedBlockClientId ) === 'default',
+			};
 		},
-		[ hasAnyBlockControls ]
+		[]
 	);
+
+	const hasAnyBlockControls = useHasAnyBlockControls();
+
+	if (
+		! isToolbarEnabled ||
+		( ! isDefaultEditingMode && ! hasAnyBlockControls )
+	) {
+		return false;
+	}
+
+	return true;
```

Comment:
```suggestion
if (isToolbarEnabled || (isDefaultEditingMode && hasAnyBlockControls)) {
    return true;
}

return false;
```

I'm wondering if we can make this easier to read by avoiding the `!`?

Reply:
I agree, but this is how the condition has been for a long time. I'd rather this be changed in a follow-up PR to avoid regression.